docs: Added documentation for automatic TCP and UDP ports waiting (#559)
kurtosis-tech/kurtosis: docs/docs/starlark-reference/port-spec.md

    #  "postgresql" to get a URL of "postgresql://..."
    # OPTIONAL (DEFAULT:"http")
    application_protocol = "http",
    

>     # All the service's TCP and UDP ports are automatically check, Kurtosis check if the declared ports

Reviewer: ```suggestion
    # Kurtosis will automatically perform a check to ensure all declared UDP and TCP ports are open and ready for traffic and connections upon startup.
```

>     # are open before finishing the service startup.

Reviewer: ```suggestion
```

>     # We can set a custom wait time out value with or disable the feature or through this property

Reviewer: ```suggestion
    # You may specify a custom wait timeout duration or disable the feature entirely.
```

>     # You can specify a custom wait time out using a Duration string:

Reviewer: ```suggestion
    # You may specify a custom wait timeout duration with a string:
```

    #  wait = "2m"

>     # Or, you can disable this feature by setting the value in None:

Reviewer: ```suggestion
    # Or, you can disable this feature by setting the value to None:
```

    #  wait = None

>     # The feature is enabled by default with a default timeout

Reviewer: ```suggestion
    # The feature is enabled by default with a default timeout of 15s
```

    # OPTIONAL (DEFAULT:"15s")
    wait = "4s"
)
```
The above constructor returns a `PortSpec` object that defines information about a port for use in [`add_service`](../concepts-reference/subnetworks.md).

> The wait field represents the timeout for the automatic ports opening check, which is the default way to check for readiness. There are others ways to check service readiness, you can use the [ServiceConfig.ReadyConditions][ready-conditions] which can be used to check if a service is ready through a REST call for instance, or you can use the [Wait][wait] instructions if you need a one-off check for readiness not attached to a service startup.

Reviewer: ```suggestion
The `wait` field represents the timeout duration that Kurtosis will use when checking whether or not a service's declared UDP or TCP port are open and ready for traffic and connections upon startup. This is the default way to perform a readiness check using Kurtosis. However, there are other ways to perform a readiness check. Specifically, you can also use [`ServiceConfig.ReadyConditions`][ready-conditions] to check if a service is ready with a a REST call, or you can use the [Wait][wait] instruction if you need to perform a one-off readiness check after service startup.
```
